### PR TITLE
Datastore TTL. Leaderboard Scoretags Multifilter

### PR DESCRIPTION
**Pull request by mofirouz**
* Warning when top-level response data is a JSON array.
* Failure when server response is not JSON
* Migrate from gameup.io to heroiclabs.com
* Add multiplayer match dequeue and queue-status calls.

### CODE REVIEW

File: GameUpSDK/Client.cs [hunk 1]
```diff
@@ -243,6 +244,46 @@ public static void Leaderboard (string id, int limit, int offset, bool withScore
...
+      if (offsetKey != null) {
+        queryParam = queryParam + "&offset_key=" + offsetKey;
+      }
+
+      if (socialIdFilters != null && socialIdFilters.Length > 0) {
+        String filters = socialIdFilters[0];
+        for (int i = 1; i < socialIdFilters.Length; i++) {
+          filters += "," + socialIdFilters[i];
+        }
+        queryParam = queryParam + "&filters=" + filters;
```
Comment (novabyte) [inline GameUpSDK/Client.cs]: Should this be `social_ids`? /cc @zyro

File: GameUpSDK/Leaderboard.cs [hunk 2]
```diff
@@ -166,7 +172,7 @@ internal Leaderboard (IDictionary<string, object> dict)
         }
       }
     }
-
+
```
Comment (novabyte) [inline GameUpSDK/Leaderboard.cs]: No need for this whitespace 😄

File: GameUpSDK/Client.cs (hunk 1 above)
Comment (zyro) [inline GameUpSDK/Client.cs]: It should, yep.

File: GameUpSDK/SessionClient.cs [hunk 3]
```diff
@@ -536,6 +575,83 @@ private void LeaderboardAndRank (string id, Boolean withScoretags, int limit, lo
...
+      if (offsetKey != null) {
+        queryParam = queryParam + "&offset_key=" + offsetKey;
+      }
+
+      if (socialIdFilters != null && socialIdFilters.Length > 0) {
+        String filters = socialIdFilters[0];
+        for (int i = 1; i < socialIdFilters.Length; i++) {
+          filters += "," + socialIdFilters[i];
+        }
+        queryParam = queryParam + "&filters=" + filters;
```
Comment (novabyte) [inline GameUpSDK/SessionClient.cs]: Same as above.

**Comment by novabyte** [comment]
@mofirouz I think instead of the new variable for `socialids` being called `filters` we should just refer to it as socialids. It keeps the implementation in line with the API reference.

**Pull request closed without merging**